knimin/lib/tests/test_data_access.py

```diff
@@ -1776,6 +1783,43 @@ def test_prepare_targeted_libraries(self):
                 self.assertEqual(obs[k], exp[k])
             self.assertEqual(obs, exp)
 
+        # testing prepare_targeted_libraries with a name
+        plate_links = [
+            {'dna_plate_id': dna_plate_ids[0], 'primer_plate_id': 1,
+             'name': 'Targeted plate 1'},
+            {'dna_plate_id': dna_plate_ids[1], 'primer_plate_id': 2,
+             'name': 'Targeted plate 2'}]
+        before = datetime.datetime.now()
+        obs_ids = db.prepare_targeted_libraries(
+            plate_links, 'test', 'ROBE', '208484Z', '108364Z', '14459',
+            'RNBD9959')
+        after = datetime.datetime.now()
+
+        for o_id in obs_ids:
+            self._clean_up_funcs.insert(
+                0, partial(db.delete_targeted_plate, o_id))
+
+        self.assertEqual(len(obs_ids), 2)
+        exp = [
+            {'id': obs_ids[0], 'name': 'Targeted plate 1', 'email': 'test',
+             'dna_plate_id': dna_plate_ids[0], 'primer_plate_id': 1,
+             'master_mix_lot': '14459', 'robot': 'ROBE',
+             'tm300_8_tool': '208484Z', 'tm50_8_tool': '108364Z',
+             'raw_concentration': None, 'mod_concentration': None,
+             'water_lot': 'RNBD9959'},
+            {'id': obs_ids[1], 'name': 'Targeted plate 2', 'email': 'test',
+             'dna_plate_id': dna_plate_ids[1], 'primer_plate_id': 2,
+             'master_mix_lot': '14459', 'robot': 'ROBE',
+             'tm300_8_tool': '208484Z', 'tm50_8_tool': '108364Z',
+             'raw_concentration': None, 'mod_concentration': None,
+             'water_lot': 'RNBD9959'}]
+        for obs_id, exp in zip(obs_ids, exp):
+            obs = db.read_targeted_plate(obs_id)
+            self.assertTrue(before <= obs.pop('created_on') <= after)
+            for k in obs:
+                self.assertEqual(obs[k], exp[k])
```

Comment:
Why is this per-key comparison needed if the following test is asserting the equality of the two objects as a whole?